### ISSUE TITLE
VM crash when setting breakpoint

### ISSUE THREAD

**Issue by jensjoha** [dart-lang/sdk#31331]
If we have a (small) file (lets call it $file_A$) with a part directive in it (to, say, $file_B$), and in this part file (i.e. $file_B$) has a field that is at a later token position than exists in $file_A$ and we try to put a breakpoint inside a class in $file_B$ (this part may or may not be necessary)  we get something like this:

```
Dumping native stack trace for thread 1e156
  [0x00000000008cdc2b] dart::TokenStream::Iterator::Advance()
  [0x00000000008cdc2b] dart::TokenStream::Iterator::Advance()
  [0x00000000008ec4f7] dart::TokenStream::Iterator::Iterator(dart::Zone*, dart::TokenStream const&, dart::TokenPosition, dart::TokenStream::Iterator::StreamType)
  [0x000000000094b268] dart::Parser::Parser(dart::Script const&, dart::Library const&, dart::TokenPosition)
  [0x000000000094f4da] dart::Parser::FieldHasFunctionLiteralInitializer(dart::Field const&, dart::TokenPosition*, dart::TokenPosition*)
  [0x0000000000867132] dart::Debugger::FindBestFit(dart::Script const&, dart::TokenPosition, dart::TokenPosition, dart::Function*)
  [0x0000000000867288] dart::Debugger::SetBreakpoint(dart::Script const&, dart::TokenPosition, dart::TokenPosition, long, long)
  [0x0000000000867c81] dart::Debugger::BreakpointLocationAtLineCol(dart::String const&, long, long)
  [0x0000000000867d1e] dart::Debugger::SetBreakpointAtLineCol(dart::String const&, long, long)
  [0x0000000000a0f8e7] Unknown symbol
  [0x0000000000a0708a] Unknown symbol
  [0x0000000000a04c81] dart::Service::InvokeMethod(dart::Isolate*, dart::Array const&, bool)
  [0x0000000000a053b7] dart::Service::HandleIsolateMessage(dart::Isolate*, dart::Array const&)
  [0x00000000008878a6] dart::IsolateMessageHandler::HandleMessage(dart::Message*)
  [0x00000000008ad10a] dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)
  [0x00000000008ad431] dart::MessageHandler::TaskCallback()
  [0x0000000000a317bd] dart::ThreadPool::Worker::Loop()
  [0x0000000000a31661] dart::ThreadPool::Worker::Main(unsigned long)
  [0x00000000009456f9] Unknown symbol
-- End of DumpStackTrace
Aborted (core dumped)
```

The error appears to be that it tries to go to the position of the field (which is in $file_B$) in the script of the top-level class (which is in $file_A$).

Service test that demonstrates is in review (https://dart-review.googlesource.com/19600)

This is only in the non-kernel version btw.

**Comment by jensjoha** [comment]
Update: As the fix seems easy I'm just going to add it in the CL.

**Comment by jensjoha** [comment]
Fixed in da75676729c444250574fd72f9cc3766e73da002.
